project/frontend/src/components/SwipeMatchForm.js

```diff
@@ -0,0 +1,188 @@
+import './SwipeMatchForm.css';
+
+import React, { useState, useEffect, useContext, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import Animals from '../scripts/animals.js';
+import { AuthContext } from './Authentication.js';
+
+import Place from '../assets/place.svg';
+import Fingerprint from '../assets/fingerprint.svg';
+import Username from '../assets/username.svg';
+
+/**
+ * A form used to gather pertinent information from a user for the 'Swipe Match'
+ * service.
+ *
+ * @param {!Object<string, number>} currLocation The current latitude and longitude
+ *     coordinates of the user.
+ * @param {string} locationName The formatted address of the current user.
+ */
+function SwipeMatchForm(props) {
+  const { currLocation, locationName } = props;
+
+  const history = useHistory();
+  const authContext = useContext(AuthContext);
+  const signedIn =
+    authContext.currentUser.get && authContext.currentUser.get.isSignedIn();
+
+  const [username, setUsername] = useState('');
+  const [groupId, setGroupId] = useState('');
+
+  const anonUsername = useRef('Anonymous ' + Animals.random());
+
+  useEffect(() => {
+    const user = authContext.currentUser.get;
+    if (user && user.isSignedIn()) {
+      setUsername(user.getBasicProfile().getName());
+    } else {
+      setUsername('');
+    }
+  }, [authContext]);
+
+  const createSession = (event) => {
+    event.preventDefault();
+    history.push({
+      pathname: '/swipe-match',
+      state: {
+        currLocation,
+        action: 'create',
+        username: username ? username : anonUsername.current,
+      },
+    });
+  };
+
+  const joinSession = (event) => {
+    event.preventDefault();
+    history.push({
+      pathname: '/swipe-match',
+      state: {
+        currLocation,
+        action: 'join',
+        username: username ? username : anonUsername.current,
+        groupId,
+      },
+    });
+  };
+
+  const signIn = (event) => {
+    event.preventDefault();
+    authContext.signIn();
+  };
+
+  return (
+    <div className='swipe-match-form-container'>
+      <form onSubmit={createSession}>
+        <h4>Create a new session.</h4>
+        <p>
+          The restaurants presented in this Swipe Match session will be based on
+          the location below. You may optionally provide futher filters for the
+          restaurants to be shown by expanding the filter dropdown. Upon
+          creation, you will be provided with a group ID which others may use to
+          join your session. You must be signed in to create a Swipe Match
+          session, though you may still use an anonymous username.
+        </p>
+        <div className='swipe-match-authentication-lock'>
+          <div style={{ opacity: signedIn ? 1 : 0.25 }}>
+            <table className='swipe-match-table'>
+              <tbody>
+                <tr>
+                  <td>
+                    <img src={Username} alt='' />
```

Comment:
Should we set these alt fields (here and below) to describe what these images are?